### PR TITLE
vscode-extensions.adzero.vscode-sievehighlight: init 1.0.6

### PR DESCRIPTION
**Pull request by SebTM**
## Description of changes

Had to reformat default.nix for extensions to follow nixfmt-rfc-style here: https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/editors/vscode/extensions#conventions-for-adding-new-extensions

Added new extension for sieve-code syntax-highlighting.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -144,6 +144,18 @@ let
...
       };
 
+      adzero.vscode-sievehighlight = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vscode-sievehighlight";
+          publisher = "adzero";
+          version = "1.0.6";
+          hash = "sha256-8Ompv792eI2kIH+5+KPL9jAf88xsMGQewHEQwi8BhoQ=";
+        };
+        meta = {
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:154]: Meta fields are missing.

**Comment by superherointj** [comment]
I've extracted the global formatting to a discrete PR: #305560

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 1 above)
Comment (SebTM) [reply pkgs/applications/editors/vscode/extensions/default.nix:152]: Sorry must have got lost when stashing to nixfmt etc. readded it again.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -142,6 +142,22 @@ let
...
+          changelog = "https://marketplace.visualstudio.com/items/adzero.vscode-sievehighlight/changelog";
+          description = " Syntax highlight support for Sieve mail filtering language.";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:154]:
```suggestion
          description = "Syntax highlight support for Sieve mail filtering language";
```
- No `.` at end of the description.
- `A Visual Studio Code extension` is missing.
- There was a space at beginning of description.

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:154]: Optional: Consider adding yourself as maintainer to be notified on updates.

**Review by superherointj — APPROVED** [review]

**Comment by SebTM** [comment]
> Can you re-fetch master and rebase? Eval is failing.

Not fixed on master - depending on https://github.com/NixOS/nixpkgs/pull/305609

**Pull request merged**